server/services/player.ts

```diff
@@ -0,0 +1,73 @@
+import { Request } from 'express';
+import { Match, Player, User } from '../models';
+import updatedPLayerSchema from '../validations/playerSchema';
+
+const getPlayerService = async (id: number) => {
+  const isPLayerExist = await Player.findOne({ where: { UserId: id } });
+
+  if (!isPLayerExist) {
+    return {
+      status: 401,
+      messege: 'هذا اللاعب غير موجود',
+    };
+  }
+
+  const player = await User.findOne({
+    where: { id },
+    attributes: ['username', 'phone', 'id'],
+    include: [
+      {
+        model: Player,
+      },
+    ],
+  });
+
+  return player;
+};
+
+const updatePlayerService = async (req: Request) => {
+  // const { UserId } = req.UserData;
+
+  const playerId = 5;
+  const { body } = req;
+  const isPLayerExist = await Player.findOne({ where: { UserId: playerId } });
+
+  if (!isPLayerExist) {
+    return {
+      status: 401,
+      data: 'هذا اللاعب غير متاح',
+    };
+  }
+  await updatedPLayerSchema.validateAsync(body);
+
+  const updatedPlayer = await Player.update(
+    { ...body },
+    {
+      where: { UserId: playerId },
+    },
+  );
+
+  return {
+    status: 200,
+    data: {
+      message: 'تم الحفظ بنجاح',
+      user: updatedPlayer,
+    },
+  };
+};
+
+const playerMatchesService = async (id: number) => {
+  const isPLayerExist = Player.findOne({ where: { UserId: id } });
+  if (!isPLayerExist) {
+    return {
+      status: 401,
+      message: 'هذا اللاعب غير موجود',
+    };
+  }
+  const playerMatches = await Match.findAll({
```

Comment:
use the privileges of Sequalize, 
to don't consume resources,
example from Alaa Code : 
``` const stadium = await User.findOne({
    where: { id },
    attributes: ['username', 'phone', 'id'],
    include: [
      {
        model: Stadium,
        include: [
          {
            model: Gallery,
            as: 'stadiumGallery',
          },
        ],
      },
      {
        model: Review,
        as: 'StadiumsReviews',
      },
    ],
  });```